rmarkdowns/Week_2.Rmd

```diff
@@ -160,66 +197,131 @@ Now let's add the lakes and rivers to our map.
 
-```{r}
-ggplot() +
-    geom_sf(data = world, fill = "lightgray") +
-    geom_sf(data = states, fill = "antiquewhite1", size = 0.25) +
-    geom_sf(data = lakes50, fill = "darkblue", color = "black") +
-    geom_sf(data = rivers50, color = "lightblue", size = 0.15) +
-    geom_sf(data = washington, fill = "antiquewhite2", color = "red")+
-    geom_text(data = states, aes(x = longitude, y = latitude, label = name), size = 5, fontface = "bold", nudge_y = states$nudge_y) +
-    coord_sf(xlim = c(-130, -110), ylim = c(40, 50), expand = FALSE) +
-    ggtitle("Washington State", subtitle = "in the Pacific Northwest Region of the United States")+
-    theme_void()
+#Amber's Challenge 7 Answers (edited existing code)
+```{r, warning=FALSE}
+# add some points while we're here
+points <- data.frame(
+  longitude = c(-87.5391, -90.4024, -81.5580),
+  latitude = c(30.4105, 32.8468, 30.7300),  
+  label = c("Lillian, AL", "Yazoo City, MS", "St. Mary's, GA")
+  )
 
+# add another layer to highlight the Mississippi river, too
+miss_river <- rivers50 %>%
+  filter(name == "Mississippi") 
+
+ggplot() +
+    geom_sf(data = world, fill = "white") +
+    geom_sf(data = states, fill = "antiquewhite", size = 0.25) +
+    geom_sf(data = alabama, fill = "#A81C1C", color = "#A81C1C")+
+    geom_sf(data = lakes50, fill = "cyan3", color = "darkcyan") +
+    geom_sf(data = rivers50, color = "cyan3", size = 0.15) +
+    geom_sf(data = railroads10, color = "goldenrod", size = 0.15, linetype = "dashed") + 
+    geom_sf(data = miss_river, color = "#4682B4", size = 20) +
+    geom_point(data = points, aes(x = longitude, y = latitude), color = "black", size = 2, shape = 18) +
+    geom_label(data = points, aes(x = longitude, y = latitude, label = label), size = 2, vjust = -.3) +
+    geom_sf_text(data = states, aes(label = name), size = 3, fontface = "bold", nudge_y = states$nudge_y) +
+    coord_sf(xlim = c(-92, -80), ylim = c(29, 36), expand = FALSE) +
+    ylab("Latitude") +
+    xlab("Longitude") +
+    ggtitle("Roll Tide!", subtitle = "Alabama (Southeast USA)")+
+    theme_bw() 
+
+# I included some questions below, as I consider my Final Challenge.
 ```
 
 # Final Challenge Preparation
 
 Think about what kinds of data you want to include in your locator map for your final project. Check out the Natural Earth [features](https://www.naturalearthdata.com/features/) to see what is available, and try to retrieve the data and add different layers to your map. You can also remove the lakes and rivers layers if you want. You can go ahead and add your locator map draft to your Final_Challenge.rmd.
 
+# QUESTIONS from Amber
+```{r}
+# I want to add a legend, but the only way I knew how got too messy. I wonder if there is a GIS-approved way (that isn't just ggplot brute force).
```

Comment:
within the geom_sf there is an argument show.legend
example: geom_sf([aes](https://ggplot2.tidyverse.org/reference/aes.html)(geometry = mid, size = AREA), show.legend = "point")
you can specify whether the legend shows the point, line, or polygon data if you have multiple within the layer

or you can use a g_legend layer, apparently, described here: https://tmieno2.github.io/R-as-GIS-for-Economists/fine-tune.html#legend

Let me know if you try these out!